From: Yaakov Selkowitz <yselkowi@redhat.com>
Date: Fri, 27 Jun 2025 09:54:08 +0000 (-0500)
Subject: Fix build with glibc-2.42
X-Git-Tag: nfs-utils-2-8-4-rc3
X-Git-Url: https://git.linux-nfs.org/?p=steved%2Fnfs-utils.git;a=commitdiff_plain;h=9f974046c37b7c28705d5558328759fff708b1cb;hp=5252588c6ac8dc6611c2bedf6b6fc86e09142d73

Fix build with glibc-2.42

exportfs.c: In function ‘release_lockfile’:
exportfs.c:83:17: error: ignoring return value of ‘lockf’ declared with attribute ‘warn_unused_result’ [-Werror=unused-result]
   83 |                 lockf(_lockfd, F_ULOCK, 0);
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~
exportfs.c: In function ‘grab_lockfile’:
exportfs.c:77:17: error: ignoring return value of ‘lockf’ declared with attribute ‘warn_unused_result’ [-Werror=unused-result]
   77 |                 lockf(_lockfd, F_LOCK, 0);
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~

lockf is now marked with attribute warn_unused_result:

https://sourceware.org/git/?p=glibc.git;a=commitdiff;h=f3c82fc1b41261f582f5f9fa12f74af9bcbc88f9

Signed-off-by: Steve Dickson <steved@redhat.com>

Origin: upstream, https://git.linux-nfs.org/?p=steved/nfs-utils.git;a=commitdiff_plain;h=9f974046c37b7c28705d5558328759fff708b1cb;hp=5252588c6ac8dc6611c2bedf6b6fc86e09142d73

---

diff --git a/utils/exportfs/exportfs.c b/utils/exportfs/exportfs.c
index b03a047..748c38e 100644
--- a/utils/exportfs/exportfs.c
+++ b/utils/exportfs/exportfs.c
@@ -74,13 +74,19 @@ grab_lockfile(void)
 {
 	_lockfd = open(lockfile, O_CREAT|O_RDWR, 0666);
 	if (_lockfd != -1)
-		lockf(_lockfd, F_LOCK, 0);
+		if (lockf(_lockfd, F_LOCK, 0) != 0) {
+			xlog_warn("%s: lockf() failed: errno %d (%s)",
+			__func__, errno, strerror(errno));
+		}
 }
 static void
 release_lockfile(void)
 {
 	if (_lockfd != -1) {
-		lockf(_lockfd, F_ULOCK, 0);
+		if (lockf(_lockfd, F_ULOCK, 0) != 0) {
+			xlog_warn("%s: lockf() failed: errno %d (%s)",
+			__func__, errno, strerror(errno));
+		}
 		close(_lockfd);
 		_lockfd = -1;
 	}
